diff --git a/debian/libdrm-intel1.symbols b/debian/libdrm-intel1.symbols
index 1ce698e..57e0110 100644
--- a/debian/libdrm-intel1.symbols
+++ b/debian/libdrm-intel1.symbols
@@ -51,11 +51,13 @@ libdrm_intel.so.1 libdrm-intel1 #MINVER#
  drm_intel_decode_set_output_file@Base 2.4.30
  drm_intel_gem_bo_aub_dump_bmp@Base 2.4.32
  drm_intel_gem_bo_clear_relocs@Base 2.4.27
+ drm_intel_gem_bo_context_exec@Base 2.4.36
  drm_intel_gem_bo_get_reloc_count@Base 2.4.27
  drm_intel_gem_bo_map_gtt@Base 2.4.3
  drm_intel_gem_bo_map_unsynchronized@Base 2.4.32
  drm_intel_gem_bo_start_gtt_access@Base 2.4.3
  drm_intel_gem_bo_unmap_gtt@Base 2.4.9
+ drm_intel_gem_bo_wait@Base 2.4.36
  drm_intel_get_aperture_sizes@Base 2.4.26
  drm_intel_get_pipe_from_crtc_id@Base 2.4.11
  drm_mmAllocMem@Base 2.4.1
