--- a/debian/libdrm-intel1.symbols (libdrm-intel1_2.4.37+git20120703.8fad46b6-0ubuntu0sarvatt_amd64)
+++ b/debian/libdrm-intel1.symbols	2012-07-03 14:47:52.954049223 -0400
@@ -58,6 +58,8 @@
  drm_intel_gem_bo_start_gtt_access@Base 2.4.3
  drm_intel_gem_bo_unmap_gtt@Base 2.4.9
  drm_intel_gem_bo_wait@Base 2.4.36
+ drm_intel_gem_context_create@Base 2.4.37
+ drm_intel_gem_context_destroy@Base 2.4.37
  drm_intel_get_aperture_sizes@Base 2.4.26
  drm_intel_get_pipe_from_crtc_id@Base 2.4.11
  drm_mmAllocMem@Base 2.4.1
