--- a/debian/libdrm2.symbols (libdrm2_2.4.35+git20120611.9e0026d3-0ubuntu0sarvatt_amd64)
+++ b/debian/libdrm2.symbols	2012-06-11 12:19:54.639325397 -0400
@@ -89,6 +89,7 @@
  drmModeFreeEncoder@Base 2.4.3
  drmModeFreeFB@Base 2.4.3
  drmModeFreeModeInfo@Base 2.4.3
+ drmModeFreeObjectProperties@Base 2.4.35
  drmModeFreePlane@Base 2.4.30
  drmModeFreePlaneResources@Base 2.4.31
  drmModeFreeProperty@Base 2.4.3
@@ -104,6 +105,8 @@
  drmModeGetPropertyBlob@Base 2.4.3
  drmModeGetResources@Base 2.4.3
  drmModeMoveCursor@Base 2.4.3
+ drmModeObjectGetProperties@Base 2.4.35
+ drmModeObjectSetProperty@Base 2.4.35
  drmModePageFlip@Base 2.4.17
  drmModeRmFB@Base 2.4.3
  drmModeSetCrtc@Base 2.4.3
