--- a/debian/libgbm1.symbols (libgbm1_8.1~git20120504.5cc4b4aa-0ubuntu0sarvatt3~precise_i386)
+++ b/debian/libgbm1.symbols	2012-05-04 18:32:53.372632088 +0000
@@ -18,6 +18,7 @@
  gbm_bo_get_user_data@Base 8.1~
  gbm_bo_get_width@Base 7.11~1
  gbm_bo_set_user_data@Base 8.1~
+ gbm_bo_write@Base 8.1~
  gbm_create_device@Base 7.11~1
  gbm_device_destroy@Base 7.11~1
  gbm_device_get_backend_name@Base 7.11~1
