diff --git a/debian/patches/13-llvm-config-pick-a-version.diff b/debian/patches/13-llvm-config-pick-a-version.diff
index 0350aa2..2893b22 100644
--- a/debian/patches/13-llvm-config-pick-a-version.diff
+++ b/debian/patches/13-llvm-config-pick-a-version.diff
@@ -13,7 +13,7 @@ Index: mesa/configure.ac
  fi
  if test "x$enable_gallium_llvm" = xyes; then
 -    AC_PATH_PROG([LLVM_CONFIG], [llvm-config], [no])
-+    AC_PATH_PROG([LLVM_CONFIG], [llvm-config-3.0], [no])
++    AC_PATH_PROG([LLVM_CONFIG], [llvm-config-3.1], [no])
  
      if test "x$LLVM_CONFIG" != xno; then
  	LLVM_VERSION=`$LLVM_CONFIG --version | sed 's/svn.*//g'`
diff --git a/debian/control b/debian/control
index 2d83129..b1bab52 100644
--- a/debian/control
+++ b/debian/control
@@ -6,7 +6,8 @@ XSBC-Original-Maintainer: Debian X Strike Force <debian-x@lists.debian.org>
 Uploaders: Cyril Brulebois <kibi@debian.org>
 Standards-Version: 3.8.4
 Build-Depends:
- debhelper (>= 8.1.3),
+ libtool,
+ debhelper (>= 8.1.2ubuntu2),
  quilt (>= 0.40),
  pkg-config,
  libdrm-dev (>= 2.4.34) [!hurd-any],
@@ -31,92 +32,12 @@ Build-Depends:
  libudev-dev [linux-any],
  flex,
  bison,
- llvm-3.0-dev [amd64 i386 kfreebsd-amd64 kfreebsd-i386],
+ llvm-3.1-dev [amd64 i386 kfreebsd-amd64 kfreebsd-i386],
  libwayland-dev (>= 0.85.0) [linux-any],
 Vcs-Git: git://git.debian.org/git/pkg-xorg/lib/mesa
 Vcs-Browser: http://git.debian.org/?p=pkg-xorg/lib/mesa.git
 Homepage: http://mesa3d.sourceforge.net/
 
-Package: libgl1-mesa-swx11
-Section: libs
-Priority: extra
-Architecture: any
-Depends:
- libosmesa6 (>= 6.5.2-1),
- ${shlibs:Depends},
- ${misc:Depends},
-Conflicts: mesag3-glide, mesag3-glide2, mesag3+ggi, libgl1, nvidia-glx, mesag3, libgl1-mesa-swrast,
- nvidia-current, fglrx
-Provides: libgl1, mesag3, libgl1-mesa-swrast
-Replaces: libgl1, mesag3, libgl1-mesa-swrast
-Pre-Depends: ${misc:Pre-Depends}
-Multi-Arch: same
-Description: free implementation of the OpenGL API -- runtime
- Mesa is a 3-D graphics library with an API which is very similar to
- that of OpenGL.  To the extent that Mesa utilizes the OpenGL command
- syntax or state machine, it is being used with authorization from
- Silicon Graphics, Inc.  However, the author makes no claim that Mesa
- is in any way a compatible replacement for OpenGL or associated with
- Silicon Graphics, Inc.
- .
- This library provides a pure software rasteriser; it does not provide
- a direct rendering-capable library, or one which uses GLX.  For that,
- please see libgl1-mesa-glx.
- .
- On Linux, this library is also known as libGL or libGL.so.1.
-
-Package: libgl1-mesa-swx11-dbg
-Section: debug
-Priority: extra
-Architecture: any
-Pre-Depends: dpkg (>= 1.15.6~)
-Depends:
- libgl1-mesa-swx11 (= ${binary:Version}),
- ${misc:Depends},
-Conflicts: libgl1-mesa-swrast-dbg
-Provides: libgl1-mesa-swrast-dbg
-Replaces: libgl1-mesa-swrast-dbg
-Multi-Arch: same
-Description: free implementation of the OpenGL API -- debugging symbols
- Mesa is a 3-D graphics library with an API which is very similar to
- that of OpenGL.  To the extent that Mesa utilizes the OpenGL command
- syntax or state machine, it is being used with authorization from
- Silicon Graphics, Inc.  However, the author makes no claim that Mesa
- is in any way a compatible replacement for OpenGL or associated with
- Silicon Graphics, Inc.
- .
- This library provides a pure software rasteriser; it does not provide
- a direct rendering-capable library, or one which uses GLX.  For that,
- please see libgl1-mesa-glx.
- .
- On Linux, this library is also known as libGL or libGL.so.1.
- .
- This package contains debugging symbols for the software rasterization GL
- library.
-
-
-Package: libgl1-mesa-swx11-dev
-Section: libdevel
-Priority: extra
-Architecture: any
-Depends:
- libgl1-mesa-swx11 (= ${binary:Version}),
- libx11-dev,
- libxext6,
- mesa-common-dev (= ${binary:Version}),
- ${misc:Depends},
-Provides: libgl-dev, mesag-dev, libgl1-mesa-swrast-dev
-Conflicts: mesa-dev, libgl-dev, mesag3 (<< 3.1-1), nvidia-glx-dev, mesag-dev, libgl1-mesa-swrast-dev
-Replaces: libgl-dev, mesag-dev, libgl1-mesa-swrast-dev
-Description: free implementation of the OpenGL API -- development files
- This package provides the development environment required for
- compiling programs with Mesa.  For a complete description of Mesa,
- please look at the libgl1-mesa-swx11 package.
- .
- This library provides a pure software rasteriser; it does not provide
- a direct rendering-capable library, or one which uses GLX.  For that,
- please see libgl1-mesa-dev.
-
 Package: libxatracker1
 Section: libs
 Architecture: linux-any 
@@ -588,6 +509,7 @@ Architecture: any
 Depends:
  ${shlibs:Depends},
  ${misc:Depends}
+Recommends: libtxc-dxtn0
 Suggests: libglide3
 Conflicts: xlibmesa-dri (<< 1:7.0.0)
 Replaces: xlibmesa-dri (<< 1:7.0.0), libgl1-mesa-dri-experimental (<< 7.11.1)
@@ -689,6 +611,14 @@ Depends:
  mesa-common-dev (= ${binary:Version}),
  libgl1-mesa-glx (= ${binary:Version}),
  libxext-dev,
+ libdrm-dev,
+ x11proto-dri2-dev,
+ x11proto-gl-dev,
+ libxdamage-dev,
+ libxfixes-dev,
+ libx11-xcb-dev,
+ libxcb-glx0-dev,
+ libxxf86vm-dev,
  ${misc:Depends},
 Conflicts: libgl-dev, libgl1-mesa-dri-dev
 Replaces: libgl-dev, libgl1-mesa-dri-dev
@@ -717,78 +647,4 @@ Description: Developer documentation for Mesa
  extensions, the complete set of release notes and the development header
  files common to all Mesa packages.
 
-Package: libosmesa6
-Section: libs
-Architecture: any
-Depends:
- ${shlibs:Depends},
- ${misc:Depends},
-Replaces: libgl1-mesa-swx11 (<< 6.5.2)
-Pre-Depends: ${misc:Pre-Depends}
-Multi-Arch: same
-Description: Mesa Off-screen rendering extension
- OSmesa is a Mesa extension that allows programs to render to an
- off-screen buffer using the OpenGL API without having to create a
- rendering context on an X Server.  It uses a pure software renderer.
- .
- This package provides both 16-bit and 32-bit versions of the off-screen
- renderer which do not require external libraries to work.
-
-Package: libosmesa6-dev
-Section: libdevel
-Architecture: any
-Depends:
- libosmesa6 (= ${binary:Version}),
- mesa-common-dev (= ${binary:Version}) | libgl-dev,
- ${misc:Depends},
-Conflicts: xlibosmesa-dev, libosmesa4-dev, libosmesa-dev
-Replaces: xlibosmesa-dev, libosmesa-dev, libgl1-mesa-swx11-dev (<< 6.5.2), mesa-common-dev (<< 6.5.2)
-Provides: xlibosmesa-dev, libosmesa-dev
-Description: Mesa Off-screen rendering extension -- development files
- This package provides the required environment for developing programs
- that use the off-screen rendering extension of Mesa.
- .
- For more information on OSmesa see the libosmesa6 package.
-
-Package: libglu1-mesa
-Section: libs
-Architecture: any
-Depends:
- ${shlibs:Depends},
- ${misc:Depends},
-Provides: libglu1
-Conflicts: mesag3 (<< 5.0.0-1), xlibmesa3, libglu1
-Replaces: libglu1
-Pre-Depends: ${misc:Pre-Depends}
-Multi-Arch: same
-Description: Mesa OpenGL utility library (GLU)
- GLU offers simple interfaces for building mipmaps; checking for the
- presence of extensions in the OpenGL (or other libraries which follow
- the same conventions for advertising extensions); drawing
- piecewise-linear curves, NURBS, quadrics and other primitives
- (including, but not limited to, teapots); tesselating surfaces; setting
- up projection matrices and unprojecting screen coordinates to world
- coordinates.
- .
- On Linux, this library is also known as libGLU or libGLU.so.1.
- .
- This package provides the SGI implementation of GLU shipped with the
- Mesa package (ergo the "-mesa" suffix).
-
-Package: libglu1-mesa-dev
-Section: libdevel
-Architecture: any
-Depends:
- libglu1-mesa (= ${binary:Version}),
- libgl1-mesa-dev | libgl-dev,
- ${misc:Depends},
-Provides: libglu-dev, xlibmesa-glu-dev
-Conflicts: mesag-dev (<< 5.0.0-1), mesa-glide2-dev (<< 5.0.0-1), mesag3+ggi-dev (<< 5.0.0-1), xlibmesa-dev
-Replaces: libglu-dev
-Description: Mesa OpenGL utility library -- development files
- Includes headers and static libraries for compiling programs with GLU.
- .
- For a complete description of GLU, please look at the libglu1-mesa
- package.
-
 # vim: tw=0
diff --git a/debian/libegl1-mesa-drivers.install.linux.in b/debian/libegl1-mesa-drivers.install.linux.in
index d370557..9bd3fde 100644
--- a/debian/libegl1-mesa-drivers.install.linux.in
+++ b/debian/libegl1-mesa-drivers.install.linux.in
@@ -3,4 +3,4 @@ dri/usr/lib/${DEB_HOST_MULTIARCH}/egl/egl_gallium.so usr/lib/${DEB_HOST_MULTIARC
 
 # Wayland support, only on Linux:
 dri/usr/lib/${DEB_HOST_MULTIARCH}/libwayland-egl.so.1 usr/lib/${DEB_HOST_MULTIARCH}/mesa-egl
-dri/usr/lib/${DEB_HOST_MULTIARCH}/libwayland-egl.so.1.0 usr/lib/${DEB_HOST_MULTIARCH}/mesa-egl
+dri/usr/lib/${DEB_HOST_MULTIARCH}/libwayland-egl.so.1.0.0 usr/lib/${DEB_HOST_MULTIARCH}/mesa-egl
diff --git a/debian/libegl1-mesa.symbols b/debian/libegl1-mesa.symbols
index cc73326..eef6138 100644
--- a/debian/libegl1-mesa.symbols
+++ b/debian/libegl1-mesa.symbols
@@ -3,6 +3,19 @@ libEGL.so.1 libegl1-mesa #MINVER# | libegl1-x11
 # drivers.  Handle the dependency explicitly in the driver
 # package.
  (regex)"^_egl.*@Base$" 7.8.1
+ dri2_add_config@Base 8.1~
+ dri2_create_image_khr@Base 8.1~
+ dri2_create_screen@Base 8.1~
+ dri2_get_device_name_for_fd@Base 8.1~
+ dri2_get_driver_for_fd@Base 8.1~
+ dri2_initialize_drm@Base 8.1~
+ dri2_initialize_wayland@Base 8.1~
+ dri2_initialize_x11@Base 8.1~
+ dri2_load_driver@Base 8.1~
+ dri2_load_driver_swrast@Base 8.1~
+ dri2_lookup_egl_image@Base 8.1~
+ dri2_setup_screen@Base 8.1~
+ dri2_to_egl_attribute_map@Base 8.1~
  eglBindAPI@Base 7.8.1
  eglBindTexImage@Base 7.8.1
  eglBindWaylandDisplayWL@Base 7.11~0
@@ -63,4 +76,11 @@ libEGL.so.1 libegl1-mesa #MINVER# | libegl1-x11
  eglWaitClient@Base 7.8.1
  eglWaitGL@Base 7.8.1
  eglWaitNative@Base 7.8.1
+ image_lookup_extension@Base 8.1~
+ use_invalidate@Base 8.1~
+ wayland_buffer_is_drm@Base 8.1~
+ wayland_drm_buffer_get_buffer@Base 8.1~
+ wayland_drm_buffer_get_format@Base 8.1~
+ wayland_drm_init@Base 8.1~
+ wayland_drm_uninit@Base 8.1~
  (arch=linux-any)wl_drm_interface@Base 8.0-2~
diff --git a/debian/libgbm1.symbols b/debian/libgbm1.symbols
index ea06a9e..9323077 100644
--- a/debian/libgbm1.symbols
+++ b/debian/libgbm1.symbols
@@ -1,16 +1,32 @@
 libgbm.so.1 libgbm1 #MINVER#
+ _gbm_create_device@Base 8.1~
  _gbm_fd_get_device_name@Base 7.11~1
  _gbm_log@Base 7.11~1
  _gbm_mesa_get_device@Base 7.11~1
  _gbm_udev_device_new_from_fd@Base 7.11~1
+ devices@Base 8.1~
+ dri2_loader_extension@Base 8.1~
+ dri_fd_get_driver_name@Base 8.1~
  gbm_bo_create@Base 7.11~1
  gbm_bo_create_from_egl_image@Base 7.11~1
  gbm_bo_destroy@Base 7.11~1
+ gbm_bo_get_device@Base 8.1~
+ gbm_bo_get_format@Base 8.1~
  gbm_bo_get_handle@Base 7.11~1
  gbm_bo_get_height@Base 7.11~1
  gbm_bo_get_pitch@Base 7.11~1
+ gbm_bo_get_user_data@Base 8.1~
  gbm_bo_get_width@Base 7.11~1
+ gbm_bo_set_user_data@Base 8.1~
+ gbm_bo_write@Base 8.1~
  gbm_create_device@Base 7.11~1
  gbm_device_destroy@Base 7.11~1
  gbm_device_get_backend_name@Base 7.11~1
  gbm_device_get_fd@Base 7.11~1
+ gbm_device_is_format_supported@Base 8.1~
+ gbm_dri_backend@Base 8.1~
+ gbm_surface_create@Base 8.1~
+ gbm_surface_destroy@Base 8.1~
+ gbm_surface_has_free_buffers@Base 8.1~
+ gbm_surface_lock_front_buffer@Base 8.1~
+ gbm_surface_release_buffer@Base 8.1~
diff --git a/debian/libgl1-mesa-dev.install.in b/debian/libgl1-mesa-dev.install.in
index 5934ec4..79c3d4a 100644
--- a/debian/libgl1-mesa-dev.install.in
+++ b/debian/libgl1-mesa-dev.install.in
@@ -1,2 +1,2 @@
-usr/lib/${DEB_HOST_MULTIARCH}/libGL.so usr/lib/${DEB_HOST_MULTIARCH}/mesa
-usr/lib/${DEB_HOST_MULTIARCH}/pkgconfig/gl.pc
+dri/usr/lib/${DEB_HOST_MULTIARCH}/libGL.so usr/lib/${DEB_HOST_MULTIARCH}/mesa
+dri/usr/lib/${DEB_HOST_MULTIARCH}/pkgconfig/gl.pc usr/lib/${DEB_HOST_MULTIARCH}/pkgconfig
diff --git a/debian/libgl1-mesa-dri.install.in b/debian/libgl1-mesa-dri.install.in
index acc3c84..cd078ef 100644
--- a/debian/libgl1-mesa-dri.install.in
+++ b/debian/libgl1-mesa-dri.install.in
@@ -1,4 +1,4 @@
 # If this part gets modified, make sure to update .install.$os as well:
 build/dri/${DEB_HOST_MULTIARCH}/*_dri.so  usr/lib/${DEB_HOST_MULTIARCH}/dri
-build/dri/${DEB_HOST_MULTIARCH}/libglsl.so usr/lib/${DEB_HOST_MULTIARCH}/dri
-build/dri/${DEB_HOST_MULTIARCH}/libdricore.so usr/lib/${DEB_HOST_MULTIARCH}/dri
+build/dri/${DEB_HOST_MULTIARCH}/libdricore*.so.1 usr/lib/${DEB_HOST_MULTIARCH}
+dri/etc/drirc etc
diff --git a/debian/libgl1-mesa-dri.install.linux.in b/debian/libgl1-mesa-dri.install.linux.in
index 7e62d9e..f4dc45e 100644
--- a/debian/libgl1-mesa-dri.install.linux.in
+++ b/debian/libgl1-mesa-dri.install.linux.in
@@ -1,10 +1,9 @@
 # Keep this part in sync with libgl1-mesa-dri.install:
 build/dri/${DEB_HOST_MULTIARCH}/*_dri.so  usr/lib/${DEB_HOST_MULTIARCH}/dri
-build/dri/${DEB_HOST_MULTIARCH}/libglsl.so usr/lib/${DEB_HOST_MULTIARCH}/dri
-build/dri/${DEB_HOST_MULTIARCH}/libdricore.so usr/lib/${DEB_HOST_MULTIARCH}/dri
+build/dri/${DEB_HOST_MULTIARCH}/libdricore*.so.1 usr/lib/${DEB_HOST_MULTIARCH}
+dri/etc/drirc etc
 
 # This part is Linux-specific:
-build/dri/${DEB_HOST_MULTIARCH}/libgallium.so usr/lib/${DEB_HOST_MULTIARCH}/dri
 build/dri/${DEB_HOST_MULTIARCH}/gallium/r300_dri.so usr/lib/${DEB_HOST_MULTIARCH}/dri
 build/dri/${DEB_HOST_MULTIARCH}/gallium/r600_dri.so usr/lib/${DEB_HOST_MULTIARCH}/dri
 build/dri/${DEB_HOST_MULTIARCH}/gallium/nouveau_dri.so usr/lib/${DEB_HOST_MULTIARCH}/dri
diff --git a/debian/libgl1-mesa-swx11-dev.install.in b/debian/libgl1-mesa-swx11-dev.install.in
index a5f58c8..23c3b12 100644
--- a/debian/libgl1-mesa-swx11-dev.install.in
+++ b/debian/libgl1-mesa-swx11-dev.install.in
@@ -1,2 +1,2 @@
-usr/lib/${DEB_HOST_MULTIARCH}/libGL.a  usr/lib/${DEB_HOST_MULTIARCH}/mesa/
-usr/lib/${DEB_HOST_MULTIARCH}/libGL.so usr/lib/${DEB_HOST_MULTIARCH}/mesa/
+#usr/lib/${DEB_HOST_MULTIARCH}/libGL.a  usr/lib/${DEB_HOST_MULTIARCH}/mesa/
+build/dri/${DEB_HOST_MULTIARCH}/libGL.so usr/lib/${DEB_HOST_MULTIARCH}/mesa/
diff --git a/debian/libgl1-mesa-swx11.install.in b/debian/libgl1-mesa-swx11.install.in
index 237ae83..db5d404 100644
--- a/debian/libgl1-mesa-swx11.install.in
+++ b/debian/libgl1-mesa-swx11.install.in
@@ -1 +1 @@
-usr/lib/${DEB_HOST_MULTIARCH}/libGL.so.* usr/lib/${DEB_HOST_MULTIARCH}/mesa/
+build/dri/${DEB_HOST_MULTIARCH}/libGL.so.* usr/lib/${DEB_HOST_MULTIARCH}/mesa/
diff --git a/debian/libgles2-mesa.symbols b/debian/libgles2-mesa.symbols
index 034eb3b..023fead 100644
--- a/debian/libgles2-mesa.symbols
+++ b/debian/libgles2-mesa.symbols
@@ -104,6 +104,7 @@ libGLESv2.so.2 libgles2-mesa #MINVER# | libgles2
  glPixelStorei@Base 7.8.1
  glPolygonOffset@Base 7.8.1
  glProgramBinaryOES@Base 7.8.1
+ glReadBufferNV@Base 8.1~
  glReadPixels@Base 7.8.1
  glReleaseShaderCompiler@Base 7.8.1
  glRenderbufferStorage@Base 7.8.1
diff --git a/debian/not-installed b/debian/not-installed
index 869f017..273a382 100644
--- a/debian/not-installed
+++ b/debian/not-installed
@@ -1,24 +1,7 @@
 # Keep track of files we don't install:
 
 # Common list:
-NOT_INSTALLED := \
-	dri/usr/include/GL/glu.h \
-	dri/usr/include/GL/glu_mangle.h \
-	dri/usr/include/GL/osmesa.h \
-	dri/usr/include/GL/vms_x_fix.h \
-	dri/usr/include/GL/wglext.h \
-	dri/usr/include/GL/wmesa.h \
-	dri/usr/lib/*/libGL.so \
-	dri/usr/lib/*/pkgconfig/gl.pc \
-	usr/include/GL/glext.h \
-	usr/include/GL/gl.h \
-	usr/include/GL/gl_mangle.h \
-	usr/include/GL/glxext.h \
-	usr/include/GL/glx.h \
-	usr/include/GL/glx_mangle.h \
-	usr/include/GL/vms_x_fix.h \
-	usr/include/GL/wglext.h \
-	usr/include/GL/wmesa.h
+NOT_INSTALLED :=
 
 # Same for linux, kfreebsd, and hurd on i386:
 NOT_INSTALLED_kfreebsd-i386 = $(NOT_INSTALLED_i386)
diff --git a/debian/rules b/debian/rules
index f0326a3..96278ff 100755
--- a/debian/rules
+++ b/debian/rules
@@ -37,15 +37,7 @@ endif
 include debian/not-installed
 
 # build the following configurations by default
-CONFIGS = swx11+glu \
-	  swx11+glu-static \
-	  dri \
-	  osmesa \
-	  osmesa-static \
-	  osmesa16 \
-	  osmesa16-static \
-	  osmesa32 \
-	  osmesa32-static
+CONFIGS = dri
 
 STAMP_DIR = debian/stamp
 STAMP = $(STAMP_DIR)/$(DEB_BUILD_GNU_TYPE)
@@ -97,7 +89,7 @@ else
 
   # LLVM is required for r300g and recommended for swrastg on x86:
   ifneq (,$(filter $(DEB_HOST_ARCH),amd64 i386 kfreebsd-amd64 kfreebsd-i386))
-	confflags_GALLIUM += --enable-gallium-llvm
+	confflags_GALLIUM += --enable-gallium-llvm  
   endif
 
 	confflags_DIRECT_RENDERING = --enable-driglx-direct
@@ -192,12 +184,11 @@ confflags-swx11+glu-i386-i686 = \
 	CFLAGS="$(CFLAGS) -march=i686" \
 	CXXFLAGS="$(CXXFLAGS) -march=i686"
 
-autotools_files = bin/config.guess bin/config.sub
+#autotools_files = bin/config.guess bin/config.sub
 
 configure: $(QUILT_STAMPFN) configure.ac
 	for file in $(autotools_files); do cp $$file $$file.pristine; done
 	autoreconf -vfi
-
 # list the configurations that will built
 configs:
 	@echo Building the following configurations: $(CONFIGS)
@@ -308,8 +299,10 @@ binary-arch: install
 	set -e; for file in $(NOT_INSTALLED_EITHER); do rm -f debian/tmp/$$file; done
 	# No need for this .so, libglapi is just a convenience library:
 	rm debian/tmp/dri/usr/lib/${DEB_HOST_MULTIARCH}/libglapi.so
+	# automake-ification in 8.1 now adds numerious .la files
+	find debian/tmp -name '*.la' -delete
 
-	dh_install -s --fail-missing
+	dh_install -s
 
 # If the llvmpipe driver was built, copy it to the package dir
 ifneq (,$(filter swrast,$(GALLIUM_DRIVERS)))
@@ -320,12 +313,9 @@ endif
 	# Create an ld.so.conf which says where to find libGL from Mesa
 	echo "/usr/lib/$(DEB_HOST_MULTIARCH)/mesa" \
 	> $(CURDIR)/debian/libgl1-mesa-glx/usr/lib/$(DEB_HOST_MULTIARCH)/mesa/ld.so.conf
-	echo "/usr/lib/$(DEB_HOST_MULTIARCH)/mesa" \
-	> $(CURDIR)/debian/libgl1-mesa-swx11/usr/lib/$(DEB_HOST_MULTIARCH)/mesa/ld.so.conf
 
 	# Empty directory for the alternative
 	mkdir -p $(CURDIR)/debian/libgl1-mesa-glx/usr/lib/$(DEB_HOST_MULTIARCH)/xorg/x11-extra-modules
-	mkdir -p $(CURDIR)/debian/libgl1-mesa-swx11/usr/lib/$(DEB_HOST_MULTIARCH)/xorg/x11-extra-modules
 
 	# Create an ld.so.conf which says where to find libEGL, libGLES{1,2} 
 	# and libOpenVG from Mesa.
@@ -359,7 +349,7 @@ ifeq (,$(filter sparc sparc64,$(DEB_HOST_ARCH)))
 			-edebian/$$PACKAGE/usr/lib/$(DEB_HOST_MULTIARCH)/mesa-egl/\* \
 		; done
 
-	dh_makeshlibs -s --remaining-packages -- -c4
+	dh_makeshlibs -s --remaining-packages -- -c1
 else
 	# Temporarily work around FTBFS on sparc, ignore new symbols:
 	dh_makeshlibs -s -- -c1
