--- a/debian/libegl1-mesa.symbols (libegl1-mesa_7.12.0~git20111222.7fd8dc3a-0ubuntu0sarvatt_amd64)
+++ b/debian/libegl1-mesa.symbols	2011-12-22 19:26:46.000000000 +0000
@@ -37,6 +37,7 @@
  eglGetSyncAttribKHR@Base 7.9
  eglInitialize@Base 7.8.1
  eglMakeCurrent@Base 7.8.1
+ eglPostSubBufferNV@Base 7.12.0~
  eglQueryAPI@Base 7.8.1
  eglQueryContext@Base 7.8.1
  eglQueryModeStringMESA@Base 7.8.1
  
diff -u mesa-8.0.0~git20120120.7a1e941e/debian/control mesa-8.0.0~git20120120.7a1e941e/debian/control
--- mesa-8.0.0~git20120120.7a1e941e/debian/control
+++ mesa-8.0.0~git20120120.7a1e941e/debian/control
@@ -23,6 +23,7 @@
  libxext-dev,
  autoconf,
  automake,
+ libtool,
  x11proto-dri2-dev (>= 2.1),
  linux-libc-dev (>= 2.6.31) [linux-any],
  libx11-xcb-dev,
