diff --git a/src/glsl/Makefile b/src/glsl/Makefile
index 3cf9fc9..872677e 100644
--- a/src/glsl/Makefile
+++ b/src/glsl/Makefile
@@ -11,19 +11,22 @@ GLSL_SRCDIR=.
 include Makefile.sources
 
 GLCPP_SOURCES = \
-	$(LIBGLCPP_GENERATED_FILES) \
+	glcpp/glcpp-lex.c \
+	glcpp/glcpp-parse.c \
 	$(LIBGLCPP_FILES) \
 	ralloc.c \
 	glcpp/glcpp.c
 
 C_SOURCES = \
-	$(LIBGLCPP_GENERATED_FILES) \
+	glcpp/glcpp-lex.c \
+	glcpp/glcpp-parse.c \
 	$(LIBGLCPP_FILES) \
 	$(LIBGLSL_FILES)
 
 # common sources for builtin_compiler and libglsl
 CXX_SOURCES = \
-	$(BUILTIN_COMPILER_GENERATED_CXX_FILES) \
+	glsl_lexer.cpp \
+	glsl_parser.cpp \
 	$(LIBGLSL_CXX_FILES)
 
 LIBS = \
diff --git a/src/glsl/Makefile.sources b/src/glsl/Makefile.sources
index 0425fa3..70edc04 100644
--- a/src/glsl/Makefile.sources
+++ b/src/glsl/Makefile.sources
@@ -8,8 +8,8 @@ LIBGLCPP_FILES = \
 	$(GLSL_SRCDIR)/glcpp/pp.c
 
 LIBGLCPP_GENERATED_FILES = \
-	$(GLSL_SRCDIR)/glcpp/glcpp-lex.c \
-	$(GLSL_SRCDIR)/glcpp/glcpp-parse.c
+	$(TOP)/src/glsl/glcpp/glcpp-lex.c \
+	$(TOP)/src/glsl/glcpp/glcpp-parse.c
 
 # libglsl
 
@@ -101,10 +101,10 @@ BUILTIN_COMPILER_CXX_FILES = \
 	$(GLSL_SRCDIR)/builtin_stubs.cpp
 
 BUILTIN_COMPILER_GENERATED_CXX_FILES = \
-	$(GLSL_SRCDIR)/glsl_lexer.cpp \
-	$(GLSL_SRCDIR)/glsl_parser.cpp
+	$(TOP)/src/glsl/glsl_lexer.cpp \
+	$(TOP)/src/glsl/glsl_parser.cpp
 
 # libglsl generated sources
 LIBGLSL_GENERATED_CXX_FILES = \
-	$(GLSL_SRCDIR)/$(BUILTIN_COMPILER_GENERATED_CXX_FILES) \
-	$(GLSL_SRCDIR)/builtin_function.cpp
+	$(BUILTIN_COMPILER_GENERATED_CXX_FILES) \
+	$(TOP)/src/glsl/builtin_function.cpp
diff --git a/src/glx/Makefile.am b/src/glx/Makefile.am
index 37a938d..f12d06e 100644
--- a/src/glx/Makefile.am
+++ b/src/glx/Makefile.am
@@ -37,6 +37,8 @@ AM_CFLAGS = \
 	-I$(top_srcdir)/include/GL/internal \
 	-I$(top_srcdir)/src/mapi \
 	-I$(top_srcdir)/src/mapi/glapi \
+	-I$(top_builddir)/src/mapi \
+	-I$(top_builddir)/src/mapi/glapi \
 	$(SHARED_GLAPI_CFLAGS) \
 	$(EXTRA_DEFINES_XF86VIDMODE) \
 	-D_REENTRANT \
diff --git a/src/mapi/glapi/gen/Makefile.am b/src/mapi/glapi/gen/Makefile.am
index 52aeb3a..499ec09 100644
--- a/src/mapi/glapi/gen/Makefile.am
+++ b/src/mapi/glapi/gen/Makefile.am
@@ -176,7 +176,7 @@ $(XORG_GLAPI_DIR)/%.h: $(MESA_GLAPI_DIR)/%.h
 
 $(MESA_GLAPI_DIR)/glapi_mapi_tmp.h: $(MESA_MAPI_DIR)/mapi_abi.py $(COMMON_ES)
 	$(PYTHON_GEN) $< \
-		--printer glapi --mode lib gl_and_es_API.xml > $@
+		--printer glapi --mode lib $(srcdir)/gl_and_es_API.xml > $@
 
 $(MESA_GLAPI_DIR)/glprocs.h: gl_procs.py $(COMMON)
 	$(PYTHON_GEN) $< -f $(srcdir)/gl_API.xml > $@
diff --git a/src/mapi/shared-glapi/Makefile.am b/src/mapi/shared-glapi/Makefile.am
index 9485683..85a3677 100644
--- a/src/mapi/shared-glapi/Makefile.am
+++ b/src/mapi/shared-glapi/Makefile.am
@@ -1,6 +1,6 @@
 # Used by OpenGL ES or when --enable-shared-glapi is specified
 
-TOP = $(top_srcdir)
+TOP = $(top_builddir)
 GLAPI = $(top_srcdir)/src/mapi/glapi
 include $(top_srcdir)/src/mapi/mapi/sources.mak
 
diff --git a/src/mesa/Makefile.am b/src/mesa/Makefile.am
index bada760..ecfb165 100644
--- a/src/mesa/Makefile.am
+++ b/src/mesa/Makefile.am
@@ -42,8 +42,8 @@ main/git_sha1.h: main/git_sha1.h.tmp
 	fi
 
 # include glapi_gen.mk for generating glapi headers for GLES
-TOP = $(top_srcdir)
-GLAPI = $(TOP)/src/mapi/glapi/gen
+TOP = $(top_builddir)
+GLAPI = $(top_srcdir)/src/mapi/glapi/gen
 include $(GLAPI)/glapi_gen.mk
 
 BUILT_SOURCES = \
diff --git a/src/mesa/Makefile.old b/src/mesa/Makefile.old
index 4ea70d4..269b42d 100644
--- a/src/mesa/Makefile.old
+++ b/src/mesa/Makefile.old
@@ -1,6 +1,7 @@
 # src/mesa/Makefile
 
 TOP = ../..
+top_builddir = $(TOP)
 include $(TOP)/configs/current
 
 MESA_LIBS := libmesa.a libmesagallium.a
diff --git a/src/mesa/libdricore/Makefile.am b/src/mesa/libdricore/Makefile.am
index 26d8a88..bc8873b 100644
--- a/src/mesa/libdricore/Makefile.am
+++ b/src/mesa/libdricore/Makefile.am
@@ -19,7 +19,7 @@
 # FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
 # IN THE SOFTWARE.
 
-TOP=$(top_srcdir)
+TOP=$(top_builddir)
 SRCDIR = $(srcdir)/..
 include ../sources.mak
 GLSL_SRCDIR = $(srcdir)/../../glsl
@@ -28,6 +28,7 @@ include ../../glsl/Makefile.sources
 noinst_PROGRAMS =
 
 AM_CPPFLAGS = \
+	-I$(top_builddir)/src/mesa/main \
 	$(INCLUDE_DIRS) \
 	$(API_DEFINES) \
 	$(DEFINES)
@@ -44,7 +45,7 @@ libdricore@VERSION@_la_SOURCES = \
 	$(LIBGLSL_FILES) \
 	$(LIBGLSL_CXX_FILES) \
 	$(BUILTIN_COMPILER_GENERATED_CXX_FILES) \
-	$(top_srcdir)/src/glsl/builtin_function.cpp
+	$(top_builddir)/src/glsl/builtin_function.cpp
 libdricore@VERSION@_la_LDFLAGS = -version-number 1:0
 libdricore@VERSION@_la_LIBADD = libdricore-asm.la
 
@@ -58,14 +59,20 @@ libdricore_asm_la_CCASFLAGS = $(AM_CCASFLAGS) -DWORKAROUND_AUTOMAKE_OBJ_FILE_CON
 
 if HAVE_X86_ASM
 libdricore_asm_la_SOURCES += $(X86_FILES)
+libdricore_asm_la_CPPFLAGS = $(AM_CPPFLAGS) \
+	-I$(top_builddir)/src/mesa/x86
 endif
 
 if HAVE_X86_64_ASM
 libdricore_asm_la_SOURCES += $(X86_64_FILES)
+libdricore_asm_la_CPPFLAGS = $(AM_CPPFLAGS) \
+	-I$(top_builddir)/src/mesa/x86-64
 endif
 
 if HAVE_SPARC_ASM
 libdricore_asm_la_SOURCES += $(SPARC_FILES)
+libdricore_asm_la_CPPFLAGS = $(AM_CPPFLAGS) \
+	-I$(top_builddir)/src/mesa/sparc
 endif
 
 if HAVE_DRICORE
diff --git a/src/mesa/sources.mak b/src/mesa/sources.mak
index 16b1c39..73e3e9a 100644
--- a/src/mesa/sources.mak
+++ b/src/mesa/sources.mak
@@ -4,8 +4,8 @@ SRCDIR ?= .
 
 # this is part of MAIN_FILES
 MAIN_ES_FILES = \
-	$(SRCDIR)/main/api_exec_es1.c \
-	$(SRCDIR)/main/api_exec_es2.c
+	$(top_builddir)/src/mesa/main/api_exec_es1.c \
+	$(top_builddir)/src/mesa/main/api_exec_es2.c
 
 MAIN_FILES = \
 	$(SRCDIR)/main/api_arrayelt.c \
@@ -34,7 +34,7 @@ MAIN_FILES = \
 	$(SRCDIR)/main/drawpix.c \
 	$(SRCDIR)/main/drawtex.c \
 	$(SRCDIR)/main/enable.c \
-	$(SRCDIR)/main/enums.c \
+	$(top_builddir)/src/mesa/main/enums.c \
 	$(SRCDIR)/main/errors.c \
 	$(SRCDIR)/main/eval.c \
 	$(SRCDIR)/main/execmem.c \
@@ -249,11 +249,11 @@ STATETRACKER_FILES = \
 PROGRAM_FILES = \
 	$(SRCDIR)/program/arbprogparse.c \
 	$(SRCDIR)/program/hash_table.c \
-	$(SRCDIR)/program/lex.yy.c \
+	$(top_builddir)/src/mesa/program/lex.yy.c \
 	$(SRCDIR)/program/nvfragparse.c \
 	$(SRCDIR)/program/nvvertparse.c \
 	$(SRCDIR)/program/program.c \
-	$(SRCDIR)/program/program_parse.tab.c \
+	$(top_builddir)/src/mesa/program/program_parse.tab.c \
 	$(SRCDIR)/program/program_parse_extra.c \
 	$(SRCDIR)/program/prog_cache.c \
 	$(SRCDIR)/program/prog_execute.c \
@@ -373,15 +373,16 @@ COMMON_DRIVER_OBJECTS = $(COMMON_DRIVER_FILES:.c=.o)
 ### Other archives/libraries
 
 GLSL_LIBS = \
-	$(TOP)/src/glsl/libglsl.a
+	$(top_builddir)/src/glsl/libglsl.a
 
 
 ### Include directories
 
 INCLUDE_DIRS = \
-	-I$(TOP)/include \
-	-I$(TOP)/src/glsl \
-	-I$(TOP)/src/mesa \
-	-I$(TOP)/src/mapi \
-	-I$(TOP)/src/gallium/include \
-	-I$(TOP)/src/gallium/auxiliary
+	-I$(top_builddir)/include \
+	-I$(top_builddir)/src/glsl \
+	-I$(top_builddir)/src/mesa \
+	-I$(top_builddir)/src/mesa/main \
+	-I$(top_builddir)/src/mapi \
+	-I$(top_builddir)/src/gallium/include \
+	-I$(top_builddir)/src/gallium/auxiliary
