--- a/debian/libpixman-1-0.symbols (libpixman-1-0_0.19.1+git20100617.445eb638-0ubuntu0sarvatt_i386)
+++ b/debian/libpixman-1-0.symbols	2010-06-17 17:56:09.192983566 -0400
@@ -31,9 +31,11 @@
  pixman_image_create_solid_fill@Base 0
  pixman_image_fill_boxes@Base 0.18.0
  pixman_image_fill_rectangles@Base 0.15.14
+ pixman_image_get_component_alpha@Base 0.19.1
  pixman_image_get_data@Base 0
  pixman_image_get_depth@Base 0
  pixman_image_get_destroy_data@Base 0.18.0
+ pixman_image_get_format@Base 0.19.1
  pixman_image_get_height@Base 0
  pixman_image_get_stride@Base 0
  pixman_image_get_width@Base 0
@@ -66,6 +68,7 @@
  pixman_region32_init_rects@Base 0.11.2
  pixman_region32_init_with_extents@Base 0.11.2
  pixman_region32_intersect@Base 0.11.2
+ pixman_region32_intersect_rect@Base 0.19.1
  pixman_region32_inverse@Base 0.11.2
  pixman_region32_n_rects@Base 0.11.2
  pixman_region32_not_empty@Base 0.11.2
@@ -88,6 +91,7 @@
  pixman_region_init_rects@Base 0
  pixman_region_init_with_extents@Base 0
  pixman_region_intersect@Base 0
+ pixman_region_intersect_rect@Base 0.19.1
  pixman_region_inverse@Base 0
  pixman_region_n_rects@Base 0
  pixman_region_not_empty@Base 0

