diff --git a/clients/cairo-util.c b/clients/cairo-util.c
index 61a053b..008cadd 100644
--- a/clients/cairo-util.c
+++ b/clients/cairo-util.c
@@ -310,7 +310,7 @@ load_jpeg(const char *filename)
 
 	jpeg_read_header(&cinfo, TRUE);
 
-	cinfo.out_color_space = JCS_EXT_BGRX;
+	cinfo.out_color_space = JCS_RGB;
 	jpeg_start_decompress(&cinfo);
 
 	stride = cairo_format_stride_for_width(CAIRO_FORMAT_RGB24,
