diff --git a/debian/libxkbcommon-dev.install b/debian/libxkbcommon-dev.install
index 55a9b31..5b584f2 100644
--- a/debian/libxkbcommon-dev.install
+++ b/debian/libxkbcommon-dev.install
@@ -1,5 +1,6 @@
 # Use no wildcards to ensure we spot any update:
 usr/include/xkbcommon/xkbcommon.h
+usr/include/xkbcommon/xkbcommon-compat.h
 usr/include/xkbcommon/xkbcommon-keysyms.h
 usr/include/xkbcommon/xkbcommon-names.h
 usr/lib/libxkbcommon.a
