diff --git a/debian/libxkbcommon-dev.install b/debian/libxkbcommon-dev
index db79b23..270161a 100644
--- a/debian/libxkbcommon-dev.install
+++ b/debian/libxkbcommon-dev.install
@@ -1,5 +1,7 @@
 # Use no wildcards to ensure we spot any update:
-usr/include/X11/extensions/XKBcommon.h
+usr/include/xkbcommon/xkbcommon.h
+usr/include/xkbcommon/xkbcommon-keysyms.h
+usr/include/xkbcommon/xkbcommon-names.h
 usr/lib/libxkbcommon.a
 usr/lib/libxkbcommon.so
 usr/lib/pkgconfig/xkbcommon.pc

