diff --git a/debian/libgl1-mesa-dri.install.in b/debian/libgl1-mesa-dri.install.in
index bb606d6..5cf3bba 100644
--- a/debian/libgl1-mesa-dri.install.in
+++ b/debian/libgl1-mesa-dri.install.in
@@ -2,5 +2,4 @@ build/dri/src/mesa/drivers/dri/common/drirc etc
 dri/usr/lib/${DEB_HOST_MULTIARCH}/dri/*_dri.so usr/lib/${DEB_HOST_MULTIARCH}/dri
 dri/usr/lib/${DEB_HOST_MULTIARCH}/gallium-pipe/*.so usr/lib/${DEB_HOST_MULTIARCH}/gallium-pipe
 dri/usr/lib/${DEB_HOST_MULTIARCH}/libdricore*.so* usr/lib/${DEB_HOST_MULTIARCH}/
-dri/usr/lib/${DEB_HOST_MULTIARCH}/libllvmradeon*.so usr/lib/${DEB_HOST_MULTIARCH}/
 dri/usr/lib/${DEB_HOST_MULTIARCH}/lib*gallium*.so* usr/lib/${DEB_HOST_MULTIARCH}/
diff --git a/debian/patches/118-dricore-gallium.patch b/debian/patches/118-dricore-gallium.patch
index cfcc54d..46ded7e 100644
--- a/debian/patches/118-dricore-gallium.patch
+++ b/debian/patches/118-dricore-gallium.patch
@@ -71,6 +71,17 @@
  	$(top_builddir)/src/gallium/auxiliary/libgallium.la \
  	$(top_builddir)/src/gallium/state_trackers/dri/drm/libdridrm.la \
  	$(top_builddir)/src/gallium/winsys/nouveau/drm/libnouveaudrm.la \
+--- a/src/gallium/targets/osmesa/Makefile.am
++++ b/src/gallium/targets/osmesa/Makefile.am
+@@ -47,7 +47,7 @@ GLAPI_LIB = $(top_builddir)/src/mapi/glapi/libglapi.la
+ endif
+ 
+ lib@OSMESA_LIB@_la_LIBADD = \
+-	$(top_builddir)/src/mesa/libmesagallium.la \
++	@MESAGALLIUM_LIBS@ \
+ 	$(top_builddir)/src/gallium/auxiliary/libgallium.la \
+ 	$(top_builddir)/src/gallium/winsys/sw/null/libws_null.la \
+ 	$(top_builddir)/src/gallium/drivers/trace/libtrace.la \
 --- a/src/gallium/targets/dri-r300/Makefile.am
 +++ b/src/gallium/targets/dri-r300/Makefile.am
 @@ -49,7 +49,7 @@
diff --git a/debian/rules b/debian/rules
index 7c1d8fc..2a95d08 100755
--- a/debian/rules
+++ b/debian/rules
@@ -78,7 +78,7 @@ else
 	DRI_DRIVERS += i915 i965
 	# Radeonsi needs LLVM, so only build it on the subset of archs
 	# where we have LLVM enabled.
-	GALLIUM_DRIVERS += radeonsi
+	#GALLIUM_DRIVERS += radeonsi
     else
 	EXTRA_SED="-e/libllvmradeon/ s/^/\#/"
     endif
@@ -116,6 +116,8 @@ confflags-dri = \
 	--enable-shared-glapi \
 	--enable-texture-float \
 	--enable-xa \
+	--disable-xvmc \
+	--disable-vdpau \
 	$(confflags_DIRECT_RENDERING) \
 	$(confflags_EGL) \
 	$(confflags_GALLIUM) \
