diff --git a/debian/libgles1-mesa.symbols b/debian/libgles1-mesa.symbols
index 0452e2b..17e4a5b 100644
--- a/debian/libgles1-mesa.symbols
+++ b/debian/libgles1-mesa.symbols
@@ -51,6 +51,7 @@ libGLESv1_CM.so.1 libgles1-mesa #MINVER# | libgles1
  glDepthRangexOES@Base 7.8.1
  glDisable@Base 7.8.1
  glDisableClientState@Base 7.8.1
+ glDiscardFramebufferEXT@Base 9.2.0~
  glDrawArrays@Base 7.8.1
  glDrawElements@Base 7.8.1
  glDrawTexfOES@Base 7.8.1
diff --git a/debian/libgles2-mesa.symbols b/debian/libgles2-mesa.symbols
index 30c7d85..6de7868 100644
--- a/debian/libgles2-mesa.symbols
+++ b/debian/libgles2-mesa.symbols
@@ -66,6 +66,7 @@ libGLESv2.so.2 libgles2-mesa #MINVER# | libgles2
  glDetachShader@Base 7.8.1
  glDisable@Base 7.8.1
  glDisableVertexAttribArray@Base 7.8.1
+ glDiscardFramebufferEXT@Base 9.2.0~
  glDrawArrays@Base 7.8.1
  glDrawArraysInstanced@Base 9.1~
  glDrawBuffers@Base 9.1~
