diff --git a/debian/libegl1-mesa.symbols b/debian/libegl1-mesa.symbols
index f64e273..7a37755 100644
--- a/debian/libegl1-mesa.symbols
+++ b/debian/libegl1-mesa.symbols
@@ -38,6 +38,7 @@ libEGL.so.1 libegl1-mesa #MINVER# | libegl1-x11
  eglGetError@Base 7.8.1
  eglGetModeAttribMESA@Base 7.8.1
  eglGetSyncAttribKHR@Base 7.9
+ eglGetSyncValuesCHROMIUM@Base 10.3.0~
  eglGetModesMESA@Base 7.8.1
  eglGetPlatformDisplayEXT@Base 10.2
  eglGetProcAddress@Base 7.8.1

