diff --git a/debian/patches/egl-platform-mir.patch b/debian/patches/egl-platform-mir.patch
index 851deca..ee795ed 100644
--- a/debian/patches/egl-platform-mir.patch
+++ b/debian/patches/egl-platform-mir.patch
@@ -663,11 +663,11 @@
  /**
   * Map --with-egl-platforms names to platform types.
 @@ -75,7 +78,8 @@
-    { _EGL_PLATFORM_DRM, "drm" },
     { _EGL_PLATFORM_FBDEV, "fbdev" },
     { _EGL_PLATFORM_NULL, "null" },
--   { _EGL_PLATFORM_ANDROID, "android" }
-+   { _EGL_PLATFORM_ANDROID, "android" },
+    { _EGL_PLATFORM_ANDROID, "android" },
+-   { _EGL_PLATFORM_HAIKU, "haiku" }
++   { _EGL_PLATFORM_HAIKU, "haiku" },
 +   { _EGL_PLATFORM_MIR, "mir" },
  };
  
@@ -744,9 +744,9 @@
 --- a/src/egl/main/egldisplay.h
 +++ b/src/egl/main/egldisplay.h
 @@ -46,6 +46,7 @@
-    _EGL_PLATFORM_FBDEV,
     _EGL_PLATFORM_NULL,
     _EGL_PLATFORM_ANDROID,
+    _EGL_PLATFORM_HAIKU,
 +   _EGL_PLATFORM_MIR,
  
     _EGL_NUM_PLATFORMS,
diff --git a/debian/control b/debian/control
index aeeb12a..2224936 100644
--- a/debian/control
+++ b/debian/control
@@ -150,7 +150,6 @@ Depends:
  ${misc:Depends},
  libudev1 [linux-any],
  libgl1-mesa-dri (= ${binary:Version}),
-Recommends: libegl1-mesa-drivers [!hurd-any]
 Provides: libegl1-x11
 Conflicts: libegl1-x11
 Replaces: libegl1-x11
@@ -185,7 +184,6 @@ Section: libdevel
 Architecture: any
 Depends:
  libegl1-mesa (= ${binary:Version}),
- libegl1-mesa-drivers (= ${binary:Version}) [!hurd-any],
  libdrm-dev (>= 2.4.52) [!hurd-any],
  x11proto-dri2-dev (>= 2.6),
  x11proto-gl-dev (>= 1.4.14),
@@ -256,7 +254,6 @@ Depends:
  ${shlibs:Depends},
  ${misc:Depends},
  libegl1-mesa (= ${binary:Version})
-Recommends: libegl1-mesa-drivers
 Provides: libwayland-egl1
 Conflicts: libwayland-egl1
 Replaces: libwayland-egl1, libegl1-mesa-drivers  (<< 10.1.0-2)
@@ -278,59 +275,6 @@ Description: implementation of the Wayland EGL platform -- debugging symbols
  This package contains the debugging symbols for the mesa implementation of the
  Wayland EGL platform
 
-Package: libopenvg1-mesa
-Section: libs
-Architecture: linux-any kfreebsd-any
-Depends:
- ${shlibs:Depends},
- ${misc:Depends},
-Provides: libopenvg1
-Conflicts: libopenvg1
-Replaces: libopenvg1
-Pre-Depends: ${misc:Pre-Depends}
-Multi-Arch: same
-Description: free implementation of the OpenVG API -- runtime
- This package contains the mesa implementation of the OpenVG 2D acceleration
- library.  OpenVG provides a device independent and vendor-neutral interface
- for sophisticated 2D graphical applications, while allowing device
- manufacturers to provide hardware acceleration on devices ranging from wrist
- watches to full microprocessor-based desktop and server machines.
-
-Package: libopenvg1-mesa-dbg
-Section: debug
-Priority: extra
-Architecture: linux-any kfreebsd-any
-Depends:
- libopenvg1-mesa (= ${binary:Version}),
- ${misc:Depends},
-Multi-Arch: same
-Description: free implementation of the OpenVG API -- debugging symbols
- This package contains the mesa implementation of the OpenVG 2D acceleration
- library.  OpenVG provides a device independent and vendor-neutral interface
- for sophisticated 2D graphical applications, while allowing device
- manufacturers to provide hardware acceleration on devices ranging from wrist
- watches to full microprocessor-based desktop and server machines.
- .
- This package contains the debugging symbols for the OpenVG library.
-
-Package: libopenvg1-mesa-dev
-Section: libdevel
-Architecture: linux-any kfreebsd-any
-Depends:
- libopenvg1-mesa (= ${binary:Version}),
- libegl1-mesa-dev,
- ${misc:Depends},
-Multi-Arch: same
-Description: free implementation of the OpenVG API -- development files
- This package contains the mesa implementation of the OpenVG 2D acceleration
- library.  OpenVG provides a device independent and vendor-neutral interface
- for sophisticated 2D graphical applications, while allowing device
- manufacturers to provide hardware acceleration on devices ranging from wrist
- watches to full microprocessor-based desktop and server machines.
- .
- This package contains the development environment required for compiling
- programs against the OpenVG 2D acceleration library.
-
 Package: libgles1-mesa
 Section: libs
 Architecture: any
diff --git a/debian/libegl1-mesa-drivers.install.in b/debian/libegl1-mesa-drivers.install.in
index 72a3603..ecdbbe5 100644
--- a/debian/libegl1-mesa-drivers.install.in
+++ b/debian/libegl1-mesa-drivers.install.in
@@ -1,2 +1,2 @@
 # This file gets tweaked in an os-specific fashion (see libegl1-mesa-drivers.install.linux.in)
-dri/usr/lib/${DEB_HOST_MULTIARCH}/egl/egl_gallium.so usr/lib/${DEB_HOST_MULTIARCH}/egl
+#dri/usr/lib/${DEB_HOST_MULTIARCH}/egl/egl_gallium.so usr/lib/${DEB_HOST_MULTIARCH}/egl
diff --git a/debian/libgbm1.install.in b/debian/libgbm1.install.in
index be214fc..44a4050 100644
--- a/debian/libgbm1.install.in
+++ b/debian/libgbm1.install.in
@@ -1,2 +1 @@
 dri/usr/lib/${DEB_HOST_MULTIARCH}/libgbm.so.1* usr/lib/${DEB_HOST_MULTIARCH}
-dri/usr/lib/${DEB_HOST_MULTIARCH}/gbm	       usr/lib/${DEB_HOST_MULTIARCH}
diff --git a/debian/rules b/debian/rules
index fdc824b..c990cce 100755
--- a/debian/rules
+++ b/debian/rules
@@ -108,7 +108,7 @@ else
   endif
 
 	confflags_DIRECT_RENDERING = --enable-driglx-direct
-	confflags_OPENVG = --enable-openvg --enable-gallium-egl
+	# confflags_OPENVG = --enable-openvg
 	confflags_GALLIUM += --enable-vdpau
 endif
 confflags_EGL = --with-egl-platforms="$(EGL_DISPLAYS)"
@@ -249,7 +244,7 @@ binary-arch: install
 	   debian/mesa-vdpau-drivers/usr/lib/${DEB_HOST_MULTIARCH}/vdpau/
   endif
 
-	dh_install -s --fail-missing
+	dh_install -s --list-missing
 
 	# Create an ld.so.conf which says where to find libGL from Mesa
 	echo "/usr/lib/$(DEB_HOST_MULTIARCH)/mesa" \
@@ -279,11 +279,11 @@ binary-arch: install
 	# Because there's no packaging problem so onerous that accomodating
 	# proprietary alternatives can't make more difficult…
 	set -e ; for PACKAGE in \
-		libegl1-mesa libgles1-mesa libgles2-mesa libopenvg1-mesa; do \
+		libegl1-mesa libgles1-mesa libgles2-mesa; do \
 			dh_makeshlibs -p$$PACKAGE -- -c4 \
 			-edebian/$$PACKAGE/usr/lib/$(DEB_HOST_MULTIARCH)/mesa-egl/\* \
 		; done
-	dh_makeshlibs -s -Xgbm_gallium_drm.so --remaining-packages -- -c4
+	dh_makeshlibs -s --remaining-packages -- -c1
 
 	dh_installdeb -s
 	dh_shlibdeps -s -l/usr/lib/$(DEB_HOST_MULTIARCH)/mesa:\
