diff --git a/debian/control b/debian/control
index 2d83129..b1bab52 100644
--- a/debian/control
+++ b/debian/control
@@ -6,7 +6,8 @@ XSBC-Original-Maintainer: Debian X Strike Force <debian-x@lists.debian.org>
 Uploaders: Cyril Brulebois <kibi@debian.org>
 Standards-Version: 3.8.4
 Build-Depends:
- debhelper (>= 8.1.3),
+ libtool,
+ debhelper (>= 8.1.3),
  quilt (>= 0.40),
  pkg-config,
  libdrm-dev (>= 2.4.34) [!hurd-any],
@@ -588,6 +509,7 @@ Architecture: any
 Depends:
  ${shlibs:Depends},
  ${misc:Depends}
+Recommends: libtxc-dxtn0
 Suggests: libglide3
 Conflicts: xlibmesa-dri (<< 1:7.0.0)
 Replaces: xlibmesa-dri (<< 1:7.0.0), libgl1-mesa-dri-experimental (<< 7.11.1)
diff --git a/debian/rules b/debian/rules
index f0326a3..96278ff 100755
--- a/debian/rules
+++ b/debian/rules
@@ -308,7 +299,7 @@ binary-arch: install
 	# purge .la files
 	find debian/tmp/ -name '*.la' -exec rm '{}' ';'
 
-	dh_install -s --fail-missing
+	dh_install -s --list-missing
 
 	# Create an ld.so.conf which says where to find libGL from Mesa
 	echo "/usr/lib/$(DEB_HOST_MULTIARCH)/mesa" \
@@ -359,7 +349,7 @@ ifeq (,$(filter sparc sparc64,$(DEB_HOST_ARCH)))
 			-edebian/$$PACKAGE/usr/lib/$(DEB_HOST_MULTIARCH)/mesa-egl/\* \
 		; done
 
-	dh_makeshlibs -s --remaining-packages -- -c4
+	dh_makeshlibs -s --remaining-packages -- -c1
 else
 	# Temporarily work around FTBFS on sparc, ignore new symbols:
 	dh_makeshlibs -s -- -c1
