diff --git a/debian/libegl1-mesa.symbols b/debian/libegl1-mesa.symbols
index 29fce1b..66045d6 100644
--- a/debian/libegl1-mesa.symbols
+++ b/debian/libegl1-mesa.symbols
@@ -19,6 +19,7 @@ libEGL.so.1 libegl1-mesa #MINVER# | libegl1-x11
  eglCreatePixmapSurface@Base 7.8.1
  eglCreateScreenSurfaceMESA@Base 7.8.1
  eglCreateSyncKHR@Base 7.9
+ eglCreateWaylandBufferFromImageWL@Base 10.2.0~
  eglCreateWindowSurface@Base 7.8.1
  eglDestroyContext@Base 7.8.1
  eglDestroyImageKHR@Base 7.8.1
diff --git a/debian/libgbm1.symbols b/debian/libgbm1.symbols
index 46bc2e2..36b9329 100644
--- a/debian/libgbm1.symbols
+++ b/debian/libgbm1.symbols
@@ -1,7 +1,4 @@
 libgbm.so.1 libgbm1 #MINVER#
- _gbm_create_device@Base 8.1~0
- _gbm_mesa_get_device@Base 7.11~1
- devices@Base 8.1~0
  gbm_bo_create@Base 7.11~1
  gbm_bo_destroy@Base 7.11~1
  gbm_bo_get_device@Base 8.1~0
@@ -18,8 +15,6 @@ libgbm.so.1 libgbm1 #MINVER#
  gbm_device_destroy@Base 7.11~1
  gbm_device_get_backend_name@Base 7.11~1
  gbm_device_get_fd@Base 7.11~1
- gbm_device_is_format_supported@Base 8.1~0
- gbm_dri_backend@Base 8.1~0
  gbm_surface_create@Base 8.1~0
  gbm_surface_destroy@Base 8.1~0
  gbm_surface_has_free_buffers@Base 8.1~0
