diff --git a/debian/control b/debian/control
index 50d88f5..57d0e25 100644
--- a/debian/control
+++ b/debian/control
@@ -42,7 +42,6 @@ Build-Depends:
  llvm-3.4-dev [amd64 i386 kfreebsd-amd64 kfreebsd-i386 armhf],
  libelf-dev [amd64 i386 kfreebsd-amd64 kfreebsd-i386 armhf],
  libwayland-dev (>= 1.2.0) [linux-any],
- libmirclient-dev [!arm64 !powerpc !ppc64 !ppc64el],
 Vcs-Git: git://git.debian.org/git/pkg-xorg/lib/mesa
 Vcs-Browser: http://git.debian.org/?p=pkg-xorg/lib/mesa.git
 Homepage: http://mesa3d.sourceforge.net/
@@ -197,7 +196,6 @@ Depends:
  libxcb-sync-dev,
  libxshmfence-dev,
  libx11-xcb-dev,
- libmirclient-dev [!arm64 !powerpc !ppc64 !ppc64el],
  libwayland-dev (>= 1.2.0),
  ${misc:Depends},
 Description: free implementation of the EGL API -- development files
diff --git a/debian/rules b/debian/rules
index e668222..25cc06d 100755
--- a/debian/rules
+++ b/debian/rules
@@ -61,7 +61,7 @@ ifeq ($(DEB_HOST_ARCH_OS), hurd)
 	DRI_DRIVERS += swrast
 else
   ifeq ($(DEB_HOST_ARCH_OS), linux)
-	confflags_DRI3 = --enable-dri3
+	confflags_DRI3 = --disable-dri3
 # Gallium drivers require libdrm-{nouveau,radeon}, only available on Linux
 	GALLIUM_DRIVERS += nouveau svga
 #
@@ -72,9 +72,6 @@ else
 	EGL_DISPLAYS += wayland
 
 # Mir isn't built on AArch64 and PPC
-    ifeq (,$(filter $(DEB_HOST_ARCH),arm64 powerpc ppc64 ppc64el))
-	EGL_DISPLAYS += mir
-    endif
     ifeq (,$(filter $(DEB_HOST_ARCH), s390 s390x))
 	DRI_DRIVERS += nouveau
     endif
@@ -239,7 +236,7 @@ binary-arch: install
 	install -m755 -d debian/libgl1-mesa-dri/usr/lib/${DEB_HOST_MULTIARCH}/dri/
 	mv debian/tmp/dri/usr/lib/${DEB_HOST_MULTIARCH}/dri/*_dri.so \
 	   debian/libgl1-mesa-dri/usr/lib/${DEB_HOST_MULTIARCH}/dri/
-	dh_install -s --fail-missing
+	dh_install -s --list-missing
 
 	# Create an ld.so.conf which says where to find libGL from Mesa
 	echo "/usr/lib/$(DEB_HOST_MULTIARCH)/mesa" \
@@ -275,10 +272,10 @@ binary-arch: install
 	# proprietary alternatives can't make more difficult…
 	set -e ; for PACKAGE in \
 		libegl1-mesa libgles1-mesa libgles2-mesa libopenvg1-mesa ; do \
-			dh_makeshlibs -p$$PACKAGE -- -c4 \
+			dh_makeshlibs -p$$PACKAGE -- -c1 \
 			-edebian/$$PACKAGE/usr/lib/$(DEB_HOST_MULTIARCH)/mesa-egl/\* \
 		; done
-	dh_makeshlibs -s --remaining-packages -- -c4
+	dh_makeshlibs -s --remaining-packages -- -c1
 
 	dh_installdeb -s
 	dh_shlibdeps -s -l/usr/lib/$(DEB_HOST_MULTIARCH)/mesa:\
diff --git a/debian/patches/04_osmesa_version.diff b/debian/patches/04_osmesa_version.diff
index 45df2cc..0755082 100644
--- a/debian/patches/04_osmesa_version.diff
+++ b/debian/patches/04_osmesa_version.diff
@@ -9,19 +9,6 @@
  
  GLAPI_LIB = $(top_builddir)/src/mapi/glapi/libglapi.la
  if HAVE_SHARED_GLAPI
-@@ -73,9 +73,9 @@
- all-local: lib@OSMESA_LIB@.la
- 	$(MKDIR_P) $(top_builddir)/$(LIB_DIR);
- 	$(MKDIR_P) $(top_builddir)/$(LIB_DIR)/gallium;
--	ln -f .libs/lib@OSMESA_LIB@.so $(top_builddir)/$(LIB_DIR)/gallium/lib@OSMESA_LIB@.so;
--	ln -f .libs/lib@OSMESA_LIB@.so.@OSMESA_VERSION@ $(top_builddir)/$(LIB_DIR)/gallium/lib@OSMESA_LIB@.so.@OSMESA_VERSION@;
--	ln -f .libs/lib@OSMESA_LIB@.so.@OSMESA_VERSION@.0.0 $(top_builddir)/$(LIB_DIR)/gallium/
-+	ln -f .libs/lib@OSMESA_LIB@.so $(top_builddir)/$(LIB_DIR)/gallium/
-+	ln -f .libs/lib@OSMESA_LIB@.so.6 $(top_builddir)/$(LIB_DIR)/gallium/
-+	ln -f .libs/lib@OSMESA_LIB@.so.6.5.3 $(top_builddir)/$(LIB_DIR)/gallium/
- endif
- 
- pkgconfigdir = $(libdir)/pkgconfig
 --- a/src/mesa/drivers/osmesa/Makefile.am
 +++ b/src/mesa/drivers/osmesa/Makefile.am
 @@ -37,7 +37,7 @@
@@ -33,16 +20,3 @@
  
  GLAPI_LIB = $(top_builddir)/src/mapi/glapi/libglapi.la
  if HAVE_SHARED_GLAPI
-@@ -54,9 +54,9 @@
- # a while by putting a link to the driver into /lib of the build tree.
- all-local: lib@OSMESA_LIB@.la
- 	$(MKDIR_P) $(top_builddir)/$(LIB_DIR);
--	ln -f .libs/lib@OSMESA_LIB@.so $(top_builddir)/$(LIB_DIR)/lib@OSMESA_LIB@.so;
--	ln -f .libs/lib@OSMESA_LIB@.so.@OSMESA_VERSION@ $(top_builddir)/$(LIB_DIR)/lib@OSMESA_LIB@.so.@OSMESA_VERSION@;
--	ln -f .libs/lib@OSMESA_LIB@.so.@OSMESA_VERSION@.0.0 $(top_builddir)/$(LIB_DIR)/
-+	ln -f .libs/lib@OSMESA_LIB@.so $(top_builddir)/$(LIB_DIR)/
-+	ln -f .libs/lib@OSMESA_LIB@.so.6 $(top_builddir)/$(LIB_DIR)/
-+	ln -f .libs/lib@OSMESA_LIB@.so.6.5.3 $(top_builddir)/$(LIB_DIR)/
- endif
- 
- pkgconfigdir = $(libdir)/pkgconfig
