diff --git a/debian/libgbm1.symbols b/debian/libgbm1.symbols
index 9a04f8b..743a6e0 100644
--- a/debian/libgbm1.symbols
+++ b/debian/libgbm1.symbols
@@ -17,13 +17,11 @@ libgbm.so.1 libgbm1 #MINVER#
  gbm_device_get_backend_name@Base 7.11~1
  gbm_device_get_fd@Base 7.11~1
  gbm_device_is_format_supported@Base 8.1~0
- gbm_dri_backend@Base 8.1~0
  gbm_surface_create@Base 8.1~0
  gbm_surface_destroy@Base 8.1~0
  gbm_surface_has_free_buffers@Base 8.1~0
  gbm_surface_lock_front_buffer@Base 8.1~0
  gbm_surface_release_buffer@Base 8.1~0
- (regex)"^_gbm_.*@Base$" 0 1
 # stupid side effect of code generation from
 # src/egl/wayland/wayland-drm/wayland-drm.xml which ought to be private.
  (arch=linux-any)wl_drm_interface@Base 0 1
diff --git a/debian/libxatracker2.symbols b/debian/libxatracker2.symbols
index c362a39..5db671e 100644
--- a/debian/libxatracker2.symbols
+++ b/debian/libxatracker2.symbols
@@ -1,5 +1,4 @@
 libxatracker.so.2 libxatracker2 #MINVER#
- driver_descriptor@Base 0
  xa_composite_allocation@Base 0
  xa_composite_check_accelerated@Base 0
  xa_composite_done@Base 0
