diff --git a/debian/libegl1-mesa.symbols b/debian/libegl1-mesa.symbols
index 233124a..89848ac 100644
--- a/debian/libegl1-mesa.symbols
+++ b/debian/libegl1-mesa.symbols
@@ -71,6 +71,7 @@ libEGL.so.1 libegl1-mesa #MINVER# | libegl1-x11
  eglSurfaceAttrib@Base 7.8.1
  eglSwapBuffers@Base 7.8.1
  eglSwapBuffersRegionNOK@Base 7.9
+ eglSwapBuffersWithDamageEXT@Base 9.2
  eglSwapInterval@Base 7.8.1
  eglTerminate@Base 7.8.1
  eglUnbindWaylandDisplayWL@Base 7.11~0
