diff --git a/debian/weston.install b/debian/weston.install
index 5f4951b..5ace35e 100644
--- a/debian/weston.install
+++ b/debian/weston.install
@@ -1,5 +1,6 @@
 usr/bin/wcap-decode
 usr/bin/weston
+usr/bin/weston-info
 usr/bin/weston-launch
 usr/bin/weston-terminal
 
@@ -14,4 +15,5 @@ usr/lib/weston/wayland-backend.so
 usr/lib/weston/x11-backend.so
 usr/lib/weston/xwayland.so
 
+usr/share/man
 usr/share/weston
